homeassistant/components/alert/entity.py

```diff
@@ -195,9 +195,12 @@ async def async_turn_on(self, **kwargs: Any) -> None:
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Async Acknowledge alert."""
-        LOGGER.debug("Acknowledged Alert: %s", self._attr_name)
-        self._ack = True
-        self.async_write_ha_state()
+        if self._can_ack:
+            LOGGER.debug("Acknowledged Alert: %s", self._attr_name)
+            self._ack = True
+            self.async_write_ha_state()
+        else:
+            LOGGER.warning("Alert Acknowledgement Blocked: %s", self._attr_name)
```

Comment:
This is not a warning. If we need to inform users in interactions, we should do so correctly by popping things in the UI and not by the logs.

It should raise an service error instead.

Reply:
I didn't realize that was a thing (first time contributing to HA). Thank you! 

Interestingly, now the UI does not flip the toggle back but I suspect that's an issue for the front end and not something I can fix here?
![raise_error_no_flip](https://github.com/user-attachments/assets/fb60f8b6-36c6-4ee4-8181-4ad6327a112f)

Reply:
Interesting, that would be a frontend issue. Could you open up a issue for that in our frontend repository? Thanks 👍

Reply:
Absolutely! Just wanted that confirmation from you that it is indeed a frontend issue before I opened it and caused confusion if it was my mistake.

Reply:
Created https://github.com/home-assistant/frontend/issues/24455